vivisect/symboliks/common.py

```diff
@@ -81,6 +81,7 @@ def __init__(self):
         self.kids          = []
         self.parents       = []
         self.cache         = {}
+        self._reduced       = False
```

Comment:
on a separate note, i've been wondering about making walkTree, reduce, and update all use a less recursive pattern... ie. using stack-based approach like so many "while todo:" loops we already do.  it will likely come with some additional ugly, but i believe it may be worthwhile.  the use of recursion is already a potential for hitting the python interp's internal recursion limit (i've hit it), and leaving the overall administrative control in the initial caller function provides some additional troubleshooting and control helps, and could potentially allow better resource control.  for example, if we wanted the caller to keep a "done" list, this additional object field (_reduced) would be unnecessary, as well as the cleanup.

just a thought.  please stew on it.